### PR TITLE
osc: allow disabling special mouse wheel behavior

### LINKED ISSUE #13096: Option to disable scrubbing though a video with the scroll wheel
Before requesting a new feature make sure it hasn't been requested yet.
https://github.com/mpv-player/mpv/labels/meta%3Afeature-request

### Expected behavior of the wanted feature
On Sept 2nd, @N-R-K [added](https://github.com/mpv-player/mpv/commit/022dbfc977c3d88cf693ce184c2f86073ae12428) and @sfan5 approved what is one of the most annoying features since Google attached tab switching to a scroll wheel a decade ago: **scrubbing through the video with a scroll wheel.** 

I have my scroll wheel configured so it changes volume when I scroll it and I'm perfectly content with that behavior. But now I randomly scrub through a video when I try to raise or lower the volume and I lose the position in a video. 

If you're going to keep this super-annoying feature, please add an option to completely disable it in a .conf file so I don't have to modify the source myself.

Thank you.

### PR DESCRIPTION
**Pull request by Dudemanguy**
Not everyone necessarily wants this feature as it can interfere with whatever they already have the scroll wheel configured to in input.conf. Add an option for this. Fixes #13096.

### CODE REVIEW

**Comment by llyyr** [comment]
I think this should include all scrolling, not just seekbar. You can currently also change the volume or sub/audio tracks by scrolling while hovering on their icons.

**Comment by N-R-K** [comment]
Do you guys think it's worth it to try and make osc keybindings *generally* configurable? I'm thinking of something similar to `input.conf` for comparison. (This is more of a general question, not saying it has to be done in *this* PR).

> I think this should include all scrolling, not just seekbar. You can currently also change the volume or sub/audio tracks by scrolling while hovering on their icons.

I don't think anyone complained about the other components - likely because the other components are smaller in size so less chance of accidental scrolls. Maybe a comma separated list of `disable_scroll` option would work, but I have a feeling that it's going to be more trouble than worth.

**Comment by llyyr** [comment]
> I don't think anyone complained about the other components

Yes, but the reason anyone would use this option is to prevent accidental inputs. Therefore in principle this should follow for all possible ways for the osc script to eat the input

File: player/lua/osc.lua [hunk 1]
```diff
@@ -2053,10 +2054,13 @@ function osc_init()
...
-        function () mp.commandv("osd-auto", "seek",  10) end
-    ne.eventresponder["wheel_down_press"] =
-        function () mp.commandv("osd-auto", "seek", -10) end
+
+    if user_opts.mousewheelseek then
+        ne.eventresponder["wheel_up_press"] =
+            function () mp.commandv("osd-auto", "seek",  10) end
+        ne.eventresponder["wheel_down_press"] =
+            function () mp.commandv("osd-auto", "seek", -10) end
+    end
```
Comment (sfan5) [inline player/lua/osc.lua:2063]: Conversely you could argue that seeking on the volume knob interferes with having wheel seek inside the video. But nobody ever complained about that :sunglasses: 

what I'm saying: this option is too specific and *if we need to* I'd rather have one that disables all wheel bindings

**Comment by Dudemanguy** [comment]
>You can currently also change the volume or sub/audio tracks by scrolling while hovering on their icons.

Oh yeah I totally forgot about that. Yeah probably it should all be configurable.

**Comment by Dudemanguy** [comment]
I changed it to `usemousewheel` which is an "all or nothing". i.e. it disables all special mouse wheel OSC or enables it all. I'm not sure if we want to take the time to do more fine-gained control (i.e. what if I want the wheel to work with switching audio tracks but not sub tracks).

**Review by N-R-K — APPROVED** [review]
Not sure if `usemousewheel` is the best name, but seems fine otherwise.

**Comment by N-R-K** [comment]
> Not sure if `usemousewheel` is the best name, but seems fine otherwise.

Just remembered that the OP in the issue mentioned that his problem was mostly with trackpad (https://github.com/mpv-player/mpv/issues/13096#issuecomment-1858319677). So `usemousewheel` is definitely not a good name. Something like `disable_scroll` or `enable_scroll` might be more appropriate.

**Comment by CogentRedTester** [comment]
Does there need to be a verb in the name? Something like `scrollcontrols=yes/no` or `scrollwheel_controls=yes/no` conveys the intention of the option fine imo. The existing options seem to follow this idea a well (`boxvideo`, `unicodeminus`, `livemarkers`, etc).

**Comment by Dudemanguy** [comment]
I literally had no idea what to call it. I'll go with scrollcontrols I suppose since that's way better.

**Review by N-R-K — APPROVED** [review]

**Pull request merged**